src/normal.cc

```diff
@@ -88,11 +88,17 @@ template<InsertMode mode>
 void enter_insert_mode(Context& context, NormalParams)
 {
     context.input_handler().insert(mode);
+
```

Comment:
The thing is that we don't want to prevent the user from modifying the buffer, just let him know that he won't be able to save it under the same name (`:w somewhere_else` should work). That way you could open a file that contains text you're interested in -e.g. a mail-, transform it using kakoune primitives and put it in your clipboard for external use, all of that with the guarantee that the original buffer won't be modified.